### ISSUE TITLE
Incorrect formatting, missing text and highlighting errors in documentation

### ISSUE THREAD

**Issue by David-Else** [hrsh7th/nvim-compe#373]
#### Describe the bug

Formatting and highlighting is wrong. It looks really bad on all Deno documents, and has the same bugs on each section of the documents.

#### To Reproduce
Steps to reproduce the behavior with the [minimal config](https://github.com/hrsh7th/nvim-compe/blob/master/misc/minimal.vim):

1. ... use the Deno language server
2. ... start typing `Deno,con` to see the docs for `Deno.connect`

This is how it looks:

![neovim](https://user-images.githubusercontent.com/12832280/118115172-875d0a80-b3e0-11eb-8c03-78b20207afdd.png)

This is how it should ideally look from VS Code:

![code](https://user-images.githubusercontent.com/12832280/118115189-8e841880-b3e0-11eb-90df-4fd8a76aea2e.png)

Note:

- large empty gap at top
- "default is" wrongly colored
- "127.0.0.1" wrongly colored
- new line after 'named' that should not be there
-  entire code block showing ``` when it should instead apply code block coloring
- code block ends early and has `@@@` inserted wrongly
- entire last line starting `requires` is missing

I am using  treesitter for my syntax highlighting and have the latest Neovim nightly and nvim-compe. It has always done this, it is not a regression.

EDIT: Is https://github.com/hrsh7th/nvim-compe/issues/260#issuecomment-808687046 related to the Markdown problem?

**Comment by folke** [comment]
Check the linked PR above to see how you can get better formatting of the documentation window.
I changed both the `hover` and `signature_help` rendering to use the `fancy_floating_markdown` instead of the simple renderer.

Would be great to have these changes applied within `compe` as well.

**Comment by hrsh7th** [comment]
@David-Else Please try `let g:markdown_fenced_languages = ['ts=typescript']`

**Comment by folke** [comment]
@hrsh7th it would still be wrong in that case.

I made some PR's upstream that were merged in the meantime that fixes that for the lsp popups.

I'm currently working on a [new PR](https://github.com/neovim/neovim/pull/14649) that consolidates all the floating window methods and will make it easier for other plugins like `compe` to use the special markdown formatting.

Once that one is merged, I'm willing to create a PR for compe

**Comment by David-Else** [comment]
It looks a bit better now, but is still messed up:
![Screenshot from 2021-06-20 13-44-22](https://user-images.githubusercontent.com/12832280/122674614-db23f600-d1cd-11eb-81b7-a88146fbe015.png)

@folke A PR to use the new Markdown preview mode would be awesome!

**Comment by folke** [comment]
@David-Else PR is ready :)
